internal/proto/internal.proto

```diff
@@ -67,4 +67,50 @@ message PrContents {
 
   minder.v1.PullRequest pr = 1;
   repeated File files = 2;
-}
\ No newline at end of file
+}
+
+message SelectorProvider {
+  // the name of the provider, e.g. github-app-jakubtestorg
+  string name = 1;
+  // the class of the provider, e.g. github-app
+  string class = 2;
+}
+
+message SelectorRepository {
+  // the full name of the repository, e.g. stacklok/minder
+  string name = 1;
+  // the provider of the repository
+  SelectorProvider provider = 2;
+
+  // is_fork is true if the repository is a fork, nil if "don't know" or rather
+  // not applicable to this provider
+  optional bool is_fork = 3;
+  // is_private is true if the repository is private, nil if "don't know" or rather
+  // not applicable to this provider
+  optional bool is_private = 4;
+
+  map<string, string> properties = 5;
```

Comment:
should this be a google.protobuf.Struct instead?

Reply:
yes!

I added a couple of tests that check property usage for other types than string/string, too